channel-email/src/main/java/com/synopsys/integration/alert/channel/email/environment/EmailEnvironmentVariableHandler.java

```diff
@@ -0,0 +1,202 @@
+        obfuscatedModel.getSmtpPort().map(String::valueOf).ifPresent(value -> builder.addVariableValue(EMAIL_PORT_KEY, value));
+        obfuscatedModel.getSmtpFrom().ifPresent(value -> builder.addVariableValue(EMAIL_FROM_KEY, value));
+        obfuscatedModel.getSmtpAuth().map(String::valueOf).ifPresent(value -> builder.addVariableValue(AUTH_REQUIRED_KEY, value));
+        obfuscatedModel.getSmtpUsername().ifPresent(value -> builder.addVariableValue(AUTH_USER_KEY, value));
+
+        if (Boolean.TRUE.equals(obfuscatedModel.getIsSmtpPasswordSet())) {
+            builder.addVariableValue(AUTH_PASSWORD_KEY, AlertConstants.MASKED_VALUE);
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    protected void saveConfiguration(EmailGlobalConfigModel configModel, EnvironmentProcessingResult processingResult) {
+        if (processingResult.hasValues()) {
+            try {
+                configAccessor.createConfiguration(configModel);
+            } catch (AlertConfigurationException ex) {
+                logger.error("Failed to create config: ", ex);
+            }
+        }
+    }
+
+    private void configureEmailSettings(EmailGlobalConfigModel configuration) {
+        environmentVariableUtility.getEnvironmentValue(EMAIL_HOST_KEY)
+            .ifPresent(configuration::setSmtpHost);
+
+        environmentVariableUtility.getEnvironmentValue(EMAIL_PORT_KEY)
+            .filter(NumberUtils::isDigits)
+            .map(NumberUtils::toInt)
+            .ifPresent(configuration::setSmtpPort);
+
+        environmentVariableUtility.getEnvironmentValue(EMAIL_FROM_KEY)
+            .ifPresent(configuration::setSmtpFrom);
+
+        environmentVariableUtility.getEnvironmentValue(AUTH_REQUIRED_KEY)
+            .map(Boolean::valueOf)
+            .ifPresent(configuration::setSmtpAuth);
+
+        environmentVariableUtility.getEnvironmentValue(AUTH_USER_KEY)
+            .ifPresent(configuration::setSmtpUsername);
+
+        environmentVariableUtility.getEnvironmentValue(AUTH_PASSWORD_KEY)
+            .ifPresent(configuration::setSmtpPassword);
+
+    }
+
+    private void configureAdditionalProperties(EmailGlobalConfigModel configuration) {
+        Map<String, String> additionalProperties = new HashMap<>();
+        for (String additionalPropertyName : OLD_ADDITIONAL_PROPERTY_KEYSET) {
+            if (environmentVariableUtility.hasEnvironmentValue(additionalPropertyName)) {
+                String javamailPropertyName = convertVariableNameToJavamailPropertyKey(additionalPropertyName);
+                String value = environmentVariableUtility.getEnvironmentValue(additionalPropertyName).orElse(null);
+                additionalProperties.put(javamailPropertyName, value);
+            }
+        }
+        configuration.setAdditionalJavaMailProperties(additionalProperties);
+    }
+
+    public static String convertVariableNameToJavamailPropertyKey(String environmentVariableName) {
```

Comment:
We generally have our public static methods towards the top of the class but we will want to verify if this method should live here and be public static.

Reply:
@psantos1113 could you weigh in on this? The only other place I see this used is in the tests for  the original class. I'm fine with making this private and creating a private method in the test to do the same functionality.